clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -204,23 +204,14 @@ public synchronized void ensureCoordinatorReady() {
                 time.sleep(retryBackoffMs);
             }
         }
+        // Since coordinator is set in a synchronized block in the group coordinator response callback,
+        // the future associated with the cooordinator request can be safely cleared here.
+        findCoordinatorFuture = null;
```

Comment:
@hachikuji Will remove the shadowing. 
On the previous comment, is there a reason why you wouldn't just clear `findCoordinatorFuture` in `GroupCoordinatorResponseHandler`? The response handler already acquires the lock for the successful case, so locking can't be an issue. The check for `findCoordinatorFuture.isDone` can then be removed from `lookupCoordinator()`, and it avoids having to follow this instance variable around the code to find where it is set and reset.